synapseclient/extensions/curator/record_based_metadata_task.py

```diff
@@ -0,0 +1,280 @@
+
+import pandas as pd
+
+from synapseclient import Synapse
+from synapseclient.models import (
+    CurationTask,
+    Grid,
+    RecordBasedMetadataTaskProperties,
+    RecordSet,
+)
+from synapseclient.services.json_schema import JsonSchemaService
+
+
+def extract_property_titles(schema_data: Dict[str, Any]) -> List[str]:
+    """
+    Extract title fields from all properties in a JSON schema.
+    Args:
+        schema_data: The parsed JSON schema data
+    Returns:
+        List of title values from the properties
+    """
+    titles = []
+
+    # Check if 'properties' exists in the schema
+    if "properties" not in schema_data:
+        return titles
+
+    properties = schema_data["properties"]
+
+    for property_name in properties.keys():
+        titles.append(property_name)
+
+    return titles
+
+
+def create_dataframe_from_titles(titles: List[str]) -> pd.DataFrame:
+    """
+    Create an empty DataFrame with the extracted titles as column names.
+    Args:
+        titles: List of title strings to use as column names
+    Returns:
+        Empty DataFrame with titles as columns
+    """
+    if not titles:
+        return pd.DataFrame()
+
+    df = pd.DataFrame(columns=titles)
+    return df
+
+
+def extract_schema_properties_from_dict(schema_data: Dict[str, Any]) -> pd.DataFrame:
+    """
+    Process a JSON schema dictionary and return a DataFrame with property titles as columns.
+    Args:
+        schema_data: The parsed JSON schema data as a dictionary
+    Returns:
+        DataFrame with property titles as columns
+    """
+    titles = extract_property_titles(schema_data)
+
+    df = create_dataframe_from_titles(titles)
+
+    return df
+
+
+def extract_schema_properties_from_web(syn: Synapse, schema_uri: str) -> pd.DataFrame:
+    """
+    Extract schema properties from a web-based JSON schema URI using Synapse.
+
+    This function retrieves a JSON schema from a web URI through the Synapse platform
+    and extracts property titles to create a DataFrame with those titles as columns.
+
+    Args:
+        syn: Authenticated Synapse client instance
+        schema_uri: URI pointing to the JSON schema resource
+
+    Returns:
+        DataFrame with property titles from the schema as column names
+
+    """
+    try:
+        org_name, schema_name, _ = schema_uri.split("-")
+    except ValueError as e:
+        raise ValueError(
+            f"Invalid schema URI format: {schema_uri}. Expected format 'org-name-schema.name.schema-version'."
+        ) from e
+
+    js = JsonSchemaService(synapse=syn)
+    schemas_list = js.list_json_schemas(organization_name=org_name)
+    if not any(schema_name == s["schemaName"] for s in schemas_list):
+        raise ValueError(
+            f"Schema URI '{schema_uri}' not found in Synapse JSON schemas."
+        )
+
+    schema = js.get_json_schema_body(json_schema_uri=schema_uri)
+    return extract_schema_properties_from_dict(schema_data=schema)

```

Comment:
More deprecated JSON Schema code in case we decide to update to the OOP model

Reply:
Updated in https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1265